infra/scripts/testnet.toml

```diff
@@ -0,0 +1,184 @@
+# Testnet node
+########################################################################################################################
+[tasks.testnet]
+dependencies = [
+    "testnet-down",
+    "testnet-init",
+    "fendermint-deps",
+    "testnet-config",
+    "testnet-up"
+]
+
+[tasks.testnet-up]
+script = """
+if [ -z $GID ]; then GID=$(id -g); fi
+if [ -z $UID ]; then UID=$(id -u); fi
+export UID
+export GID
+export CMT_PERSISTENT_PEERS=`cat $BASE_DIR/peers`
+export CHAIN_NAME=$CHAIN_NAME
+export BASE_DIR=$BASE_DIR
+docker compose -f ./infra/docker-compose.yml up -d
+"""
+
+[tasks.testnet-down]
+dependencies = [
+    "testnet-docker-compose-down",
+    "docker-network-rm",
+]
+
+[tasks.testnet-docker-compose-down]
+script = """
+export CMT_PERSISTENT_PEERS="UNDEFINED"
+if [ -z $GID ]; then GID=$(id -g); fi
+if [ -z $UID ]; then UID=$(id -u); fi
+export UID
+export GID
+docker compose -f ./infra/docker-compose.yml down
+"""
+
+[tasks.testnet-init]
+dependencies = [
+    "testnet-clear",
+    "testnet-network-create",
+    "cometbft-pull",
+    "testnet-mkdir",
+    "testnet-cometbft-init",
+    "testnet-mkdir",
+]
+
+[tasks.testnet-network-create]
+env = { "NETWORK_NAME"="${NETWORK_NAME}"}
+extend = "docker-network-create"
+
+[tasks.testnet-clear]
+script="""
+echo clearing all IPC data
+rm -rf ${BASE_DIR}
+"""
+
+[tasks.testnet-mkdir]
+script="""
+mkdir -p $BASE_DIR
+mkdir -p $BASE_DIR/out
+
+for i in $(seq 0 3); do
+    mkdir -p $BASE_DIR/node${i}
+    mkdir -p $BASE_DIR/node${i}/fendermint
+    mkdir -p $BASE_DIR/node${i}/cometbft
+done
+"""
+
+[tasks.testnet-cometbft-init]
+dependencies = ["testnet-cometbft-init-node0", "testnet-cometbft-init-node1", "testnet-cometbft-init-node2", "testnet-cometbft-init-node3"]
+
+[tasks.testnet-cometbft-init-node0]
+extend = "cometbft-init"
+env = { "CMD" = "init", "NETWORK_NAME"="${NETWORK_NAME}", "CMT_DIR" = "${BASE_DIR}/node0/cometbft", "CMT_CONTAINER_NAME" = "cometbft-node0", "FLAGS" = "-a STDOUT -a STDERR --rm"}
+
+[tasks.testnet-cometbft-init-node1]
+extend = "cometbft-init"
+env = { "CMD" = "init", "NETWORK_NAME"="${NETWORK_NAME}", "CMT_DIR" = "${BASE_DIR}/node1/cometbft", "CMT_CONTAINER_NAME" = "cometbft-node1", "FLAGS" = "-a STDOUT -a STDERR --rm"  }
+
+[tasks.testnet-cometbft-init-node2]
+extend = "cometbft-init"
+env = { "CMD" = "init", "NETWORK_NAME"="${NETWORK_NAME}", "CMT_DIR" = "${BASE_DIR}/node2/cometbft", "CMT_CONTAINER_NAME" = "cometbft-node2", "FLAGS" = "-a STDOUT -a STDERR --rm"  }
+
+[tasks.testnet-cometbft-init-node3]
+extend = "cometbft-init"
+env = { "CMD" = "init", "NETWORK_NAME"="${NETWORK_NAME}", "CMT_DIR" = "${BASE_DIR}/node3/cometbft", "CMT_CONTAINER_NAME" = "cometbft-node3", "FLAGS" = "-a STDOUT -a STDERR --rm"  }
+
+[tasks.testnet-config]
+dependencies = [
+    "testnet-script-new-genesis",
+    "testnet-script-add-peers",
+    "testnet-script-new-key",
+    "testnet-script-new-account",
+    "testnet-script-new-gateway",
+    "testnet-script-share-genesis"
+]
```

Comment:
I reckon it's these and [the node equivalent](https://github.com/consensus-shipyard/fendermint/pull/222/files#diff-c7acb1a1a82d3b2fa4b0b12e343395aee71560f17d0a888a933fc3f2ecea4c3cR56-R64) steps that benefit least from being in the `Makefile.toml` and chained with dependencies as individual tasks, as opposed to just one script that does it all.

But it's much better now that you have split it out into multiple `.toml` files, thanks for that!